src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,232 @@
+    // this.hasNote = intake.hasNote();
+
+    if (TESTING) {
+      io.setShooterWheelBottomVelocity(bottomWheelVelocity.get());
+      io.setShooterWheelTopVelocity(topWheelVelocity.get());
+      io.setAngle(angle.get());
+    } else if (autoShooter == false) {
+      io.setShooterWheelBottomVelocity(SHOOTER_IDLE_VELOCITY);
+      io.setShooterWheelTopVelocity(SHOOTER_IDLE_VELOCITY);
+    } else {
+      this.runAngleStateMachine();
+      io.setShooterWheelBottomVelocity(SHOOTER_IDLE_VELOCITY);
+      io.setShooterWheelTopVelocity(SHOOTER_IDLE_VELOCITY);
+    }
+  }
+
+  public void updateAlliance(Alliance newAlliance) {
+    this.alliance = newAlliance;
+  }
+
+  private void runAngleStateMachine() {
+    double test =
+        Field2d.getInstance()
+            .getAllianceSpeakerCenter()
+            .minus(RobotOdometry.getInstance().getEstimatedPosition())
+            .getTranslation()
+            .getNorm();
+    if (hasNote == true) {
+      if (this.inRange() == true) {
+        if (scoringPodium == true) {
+          io.setShooterWheelBottomVelocity(ShooterConstants.PODIUM_VELOCITY);
+          io.setShooterWheelTopVelocity(ShooterConstants.PODIUM_VELOCITY);
+          io.setAngle(ShooterConstants.PODIUM_ANGLE);
+        } else if (scoringSubwoofer == true) {
+          io.setShooterWheelBottomVelocity(ShooterConstants.SUBWOOFER_VELOCITY);
+          io.setShooterWheelTopVelocity(ShooterConstants.SUBWOOFER_VELOCITY);
+          io.setAngle(ShooterConstants.SUBWOOFER_ANGLE);
+        } else if (scoringAmp == true) {
+          io.setShooterWheelBottomVelocity(ShooterConstants.AMP_VELOCITY);
+          io.setShooterWheelTopVelocity(ShooterConstants.AMP_VELOCITY);
+          io.setAngle(ShooterConstants.AMP_ANGLE);
+        } else if (aiming == true) {
+          io.setShooterWheelBottomVelocity(
+              veloTreeMap.get(RobotOdometry.getInstance().getEstimatedPosition().getX()));
+          io.setShooterWheelTopVelocity(
+              veloTreeMap.get(RobotOdometry.getInstance().getEstimatedPosition().getX()));
+          io.setAngle(angleTreeMap.get(RobotOdometry.getInstance().getEstimatedPosition().getX()));
+        } else { // not aiming
+          io.setShooterWheelBottomVelocity(ShooterConstants.SHOOTER_IDLE_VELOCITY);
+          io.setShooterWheelTopVelocity(ShooterConstants.SHOOTER_IDLE_VELOCITY);
+          io.setAngle(angleTreeMap.get(test));
+        }
+      } else { // not in range
+        io.setShooterWheelBottomVelocity(ShooterConstants.SHOOTER_IDLE_VELOCITY);
+        io.setShooterWheelTopVelocity(ShooterConstants.SHOOTER_IDLE_VELOCITY);
+        io.setAngle(ShooterConstants.SHOOTER_STORAGE_ANGLE);
+      }
+    } else { // no note
+      io.setAngle(ShooterConstants.SHOOTER_STORAGE_ANGLE);
+    }
+    this.goToConstantVelocity();
+  }
+
+  public void shoot(double topWheelVelocityRPS, double bottomWheelVelocityRPS) {
```

Comment:
I believe there are two shooting related buttons (aim and shoot). This appears to be the implementation for aim; another method is needed for shoot.